frontend/src/pages/projects/screens/detail/storage/utils.ts

```diff
@@ -24,3 +27,87 @@ export const isPvcUpdateRequired = (
   getDescriptionFromK8sResource(existingPvc) !== storageData.description ||
   existingPvc.spec.resources.requests.storage !== storageData.size ||
   existingPvc.spec.storageClassName !== storageData.storageClassName;
+
+export const handleConnectedNotebooks = (
+  tempData: ClusterStorageNotebookSelection[],
+  notebooks: NotebookKind[],
+  existingPvc?: PersistentVolumeClaimKind,
+): {
+  updatedNotebooks: ClusterStorageNotebookSelection[];
+  removedNotebooks: string[];
+  newNotebooks: ClusterStorageNotebookSelection[];
+} => {
+  const updatedNotebooks: ClusterStorageNotebookSelection[] = [];
+  const removedNotebooks: string[] = [];
+  const unChangedNotebooks: ClusterStorageNotebookSelection[] = [];
+
+  notebooks.forEach((notebook) => {
+    const tempEntry = tempData.find((item) => item.name === notebook.metadata.name);
+    if (!tempEntry) {
+      removedNotebooks.push(notebook.metadata.name);
+    } else {
+      const mountPath = existingPvc
+        ? getNotebookPVCMountPathMap(notebook)[existingPvc.metadata.name]
+        : '';
+      if (mountPath !== tempEntry.mountPath.value) {
+        updatedNotebooks.push(tempEntry);
+      }
+      if (mountPath === tempEntry.mountPath.value) {
+        unChangedNotebooks.push(tempEntry);
+      }
+    }
+  });
+
+  const newNotebooks = tempData.filter(
+    (item) =>
+      !updatedNotebooks.includes(item) &&
+      !removedNotebooks.includes(item.name) &&
+      !unChangedNotebooks.includes(item),
+  );
+
+  return { updatedNotebooks, removedNotebooks, newNotebooks };
+};
+
+export const mountPathSuffix = (mountPath: string): string =>
+  mountPath.startsWith(MOUNT_PATH_PREFIX) ? mountPath.slice(MOUNT_PATH_PREFIX.length) : mountPath;
+
+export const mountPathFormat = (mountPath: string): MountPathFormat =>
+  mountPath.startsWith(MOUNT_PATH_PREFIX) ? MountPathFormat.STANDARD : MountPathFormat.CUSTOM;
+
+export const isMountPathFormat = (value: string): value is MountPathFormat =>
+  value === MountPathFormat.STANDARD || value === MountPathFormat.CUSTOM;
+
+export const validateClusterMountPath = (value: string, inUseMountPaths: string[]): string => {
+  const format = value.startsWith(MOUNT_PATH_PREFIX)
+    ? MountPathFormat.STANDARD
+    : MountPathFormat.CUSTOM;
+
+  if (!value.length && format === MountPathFormat.CUSTOM) {
+    return 'Enter a path to a model or folder.';
+  }
+
+  if (value === '/' && format === MountPathFormat.CUSTOM) {
+    return 'Enter a path to a model or folder. This path cannot point to a root folder.';
+  }
+
+  // Regex to allow empty string for Standard format
+  const regex =
+    format === MountPathFormat.STANDARD
+      ? /^(\/?[a-z0-9-]+(\/[a-z0-9-]+)*\/?|)$/
+      : /^(\/?[a-z0-9-]+(\/[a-z0-9-]+)*\/?)?$/;
+
+  if (!regex.test(value)) {
+    return 'Must consist of lowercase letters, numbers and hyphens (-). Use slashes (/) to indicate the subdirectories';
+  }
```

Comment:
Numbers are not allowed in the mount path, check the mount path field in Spawner page.
![image](https://github.com/user-attachments/assets/f4823238-b301-4615-9c6f-bd34571b3389)

Reply:
@dpanshug , there was a conversation around this in the slack channel we are now allowing numbers as well

Reply:
@pnaik1 @dpanshug Here is [the thread link](https://redhat-internal.slack.com/archives/C07CQ4R4HMY/p1732630861994889) around this point. Users will be able to use the number in the path.